akka-cluster-tools/src/multi-jvm/scala/akka/cluster/client/ClusterClientSpec.scala

```diff
@@ -63,6 +62,115 @@ object ClusterClientSpec extends MultiNodeConfig {
     }
   }
 
+  //#clientEventsListener
+  class ClientListener(targetClient: ActorRef) extends Actor {
+    override def preStart(): Unit = {
+      context.system.eventStream.subscribe(self, classOf[ContactPointChange])
+      targetClient ! GetContactPoints
+    }
+
+    def receive: Receive =
+      starting(Set.empty[ActorSelection])
+
+    def starting(contactPoints: Set[ActorSelection]): Receive = {
+      case ContactPointAdded(cp, `targetClient`) ⇒
+        context.become(starting(contactPoints + cp))
+      case ContactPointRemoved(cp, `targetClient`) ⇒
+        context.become(starting(contactPoints - cp))
+      case ContactPoints(cps) ⇒
+        context.become(started(contactPoints ++ cps))
```

Comment:
Sorry this isn't clear - I sent it too soon. My point is that the state that the client sends the requestor may be out of date by the time the requestor receives it. The event bus will issue events in its own time. 

> On 9 May 2016, at 20:15, Christopher Hunt <huntchr@gmail.com> wrote:
> 
> 1. CL subscribes to event stream
> 2. C emits event A
> 3. CL sends C a request
> 4. C receives CL's request
> 5. CL receives event A
> 6. C replies to CL
> 7. CL receives C's reply
> 
>> On 9 May 2016, at 20:00, Patrik Nordwall <notifications@github.com> wrote:
>> 
>> In akka-cluster-tools/src/multi-jvm/scala/akka/cluster/client/ClusterClientSpec.scala:
>> 
>> > +  class ClientListener(targetClient: ActorRef) extends Actor {
>> > +    override def preStart(): Unit = {
>> > +      context.system.eventStream.subscribe(self, classOf[ContactPointChange])
>> > +      targetClient ! GetContactPoints
>> > +    }
>> > +
>> > +    def receive: Receive =
>> > +      starting(Set.empty[ActorSelection])
>> > +
>> > +    def starting(contactPoints: Set[ActorSelection]): Receive = {
>> > +      case ContactPointAdded(cp, `targetClient`) ⇒
>> > +        context.become(starting(contactPoints + cp))
>> > +      case ContactPointRemoved(cp, `targetClient`) ⇒
>> > +        context.become(starting(contactPoints - cp))
>> > +      case ContactPoints(cps) ⇒
>> > +        context.become(started(contactPoints ++ cps))
>> no, I mean that normally you ignore delta events until you get the snapshot as the starting point
>> 
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly or view it on GitHub
>>